GSoC-2025/ethology ; support for any point trackers (Richard Dushime Mudahera).md

```diff
@@ -0,0 +1,249 @@
+      <td style="border: 1px solid #ddd; padding: 8px;">30</td>
+    </tr>
+    <!-- Week 11 -->
+    <tr>
+      <td style="border: 1px solid #ddd; padding: 8px;">11</td>
+      <td style="border: 1px solid #ddd; padding: 8px;">
+        <strong>Code Freeze & Testing:</strong>- Fix edge cases (e.g., occlusions).<br>- Stress-test on large videos.
+      </td>
+      <th style="border: 1px solid #ddd; padding: 8px;">Beta version released.</th>
+      <td style="border: 1px solid #ddd; padding: 8px;">30</td>
+    </tr>
+    <!-- Week 12 -->
+    <tr>
+      <td style="border: 1px solid #ddd; padding: 8px;">12</td>
+      <td style="border: 1px solid #ddd; padding: 8px;">
+        <strong>Finalization:</strong>- Polish UI/UX.<br>- Finalize tutorials and API docs.
+      </td>
+      <th style="border: 1px solid #ddd; padding: 8px;">Code merged, presentation ready.</th>
+      <td style="border: 1px solid #ddd; padding: 8px;">30</td>
+    </tr>
+    <!-- Week 13 -->
+    <tr>
+      <td style="border: 1px solid #ddd; padding: 8px;">13</td>
+      <td style="border: 1px solid #ddd; padding: 8px;">
+         - Stay active as a community contributor by addressing bugs, offering support, and enhancing features as needed.<br> - Provide assistance through forums, Zulip, and GitHub for users adopting the tool.<br> - Monitor feedback and plan follow-up in agreement with mentors.<br> - Act as a resource for new contributors and maintain a connection with mentors.
+      </td>
+      <th style="border: 1px solid #ddd; padding: 8px;">Continued Community Engagement</th>
+      <td style="border: 1px solid #ddd; padding: 8px;">30</td>
+    </tr>
+  </tbody>
+</table>
+
+### Key Milestones
+
+- Week 3: First end-to-end tracking prototype.
+- Week 6: Mid-term review with core features functional.
+- Week 9: Stretch goals implementation begins.
+- Week 12: Project submission with full test coverage.
+
+### Communication Plan
+- **Mentor Meetings:** Hold weekly video or voice calls to discuss progress, challenges, and next steps.
+- **Daily/Weekly Updates:** Share progress and ask quick questions on Zulip.
+- **Documentation & Issue Tracking:** Use GitHub Issues and project boards to track tasks and milestones.
+- **Code Reviews:** Submit regular or weekly pull requests to ensure the project stays on track and meets community standards.
+
+## Personal Statement
+
+### Past Experience
+I have built a good foundation in Python and computer vision by working on several open source projects. My contributions to NIU repositories show that I can work well with open source communities. While I am still learning machine learning and using PyTorch, I am eager to learn more. I have worked on projects that required both research-level and production-quality code. I have also collaborated with Research Software Engineers and participated in events like the Collaborations Workshop 2024, which has deepened my interest in software sustainability and research projects and tools.
```

Comment:
It would be nice if you can provide more specific examples for the points below.

> I have built a good foundation in Python and computer vision by working on several open source projects.

>  I have worked on projects that required both research-level and production-quality code.

> I have also collaborated with Research Software Engineers 

Could you mention (or link) some examples of these projects? Especially if they are not NIU tools, can you explain what were they about, what did you implement, and maybe what did you learn?

> participated in events like the Collaborations Workshop 2024

This is very nice! Can you provide more details on your experience in the workshop?